test/lib/testserver.js

```diff
@@ -50,7 +63,14 @@ TestServer.start = function () {
     init();
   }
   do {
-    server = app.listen(port);
+    server = app.listen(port)
+      .on('error', function(err) {
+        if ('EADDRINUSE' === err.code) {
+          // Just continue trying.
```

Comment:
Yes, having two blocks here isn't necessary. I just wanted to make it explicit with the comment why we're capturing and handling every possible error, except `EADDRINUSE`. It would look a bit weird otherwise. Just legibility.